api/src/models/user/user.methods.ts

```diff
@@ -105,17 +107,42 @@ export async function addConnectionToUser (this: IUserDocument, objId: string, i
       targetUser["connectionOf"][this._id] = originatorConnection;
 
       // Saves the changes
-      const originator = await UserModel.findByIdAndUpdate(this.id, this);
-      await UserModel.findByIdAndUpdate(objId, targetUser);
+      await this.save();
```

Comment:
```suggestion
      await thisUsersDocument.save();
```

Reply:
What you've done with `this` looks much cleaner then originator... let's remove the commented out originator stuff, unless I'm missing something?